website/docs/components/form/key-value-inputs/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,148 @@
+
+If more than one row is present, users can opt to delete any of the rows.
+
+![A user's cursor is hovering over the second row's delete button](/assets/components/form/key-value-inputs/key-value-inputs-ip-delete-any-row.png)
+
+## Footer
+
+The footer includes an add button, compact alert, or fieldset validation.
+
+### Add button
+
+The add button is displayed by default and, when clicked, appends a new row to collect additional information. The labeling of this button follows our [content guidelines](/content/writing-style#add-and-remove), e.g., "Add {object}."
+
+![A single secondary button with a leading plus icon with the label "Add user"](/assets/components/form/key-value-inputs/key-value-inputs-footer-add-button.png)
+
+### Maximum number of rows reached
+
+Use a compact alert to notify the user when they've reached a maximum number of rows. While the content can be customized, we recommend using the following message: "Only {maximum number} {objects} can be added at a time."
+
+This uses the compact neutral alert, so it doesn’t visually compete with other high-priority feedback, such as input errors.
+
+![A compact neutral alert that says "Limit of fifteen invites has been reached."](/assets/components/form/key-value-inputs/key-value-inputs-footer-notification.png)
+
+### Validation
+
+In addition to form-level validation, if a specific error is associated with the fieldset, it will be displayed below the add button.
+
+![Below the legend and helper text is a compact critical alert showing a fieldset error that the user has reached the maximum number of rows.](/assets/components/form/key-value-inputs/key-value-inputs-fieldset-error.png)
+
+Read more about this in our [form patterns validation documentation](/patterns/form-patterns?tab=validation).
+
+## Responsive behavior
+
+
+### Small or larger
+
+In experiences that are greater than or equal to the [medium (`md`) breakpoint](/foundations/breakpoints), the label and helper text are only visible for the first row of inputs. This reduces visual clutter by condensing the form elements.
+
+![Key value input rows with the key and value are side by side along with a delete button to the right. Only one visible label and helper text per grouping of key and value inputs. Add new row button below the group of inputs.](/assets/components/form/key-value-inputs/key-value-inputs-ip-address-desktop.png)
+
+### Less than small
+
+However, when the viewport is less than the [medium (`md`) breakpoint](/foundations/breakpoints), all field elements stack and occupy 100% of the container and display the label and helper text for their associated inputs. As the form height increases, the visible labels and helper text help orient the user as they scroll down and fill in information. If repeating each input’s helper text isn’t useful to the user, consider putting that information in the fieldset helper text instead to reduce the visual noise.
```

Comment:
[suggestion] Couple of suggestions here, I also removed the "~~~help~~~ orient" because it seemed duplicative with the "helper text", but it does technically "help" them.

```suggestion
When the viewport is less than the [medium (`md`) breakpoint](/foundations/breakpoints), all field elements stack to occupy 100% of the container width and display the label and helper text for their associated inputs. As the form height increases, the visible labels and helper text orient the user as they scroll down and fill in information. If repeating each input’s helper text isn’t useful to the user, consider placing that information in the fieldset helper text instead to reduce the visual noise.
```